frontend/src/components/node/node.jsx

```diff
@@ -34,85 +33,113 @@ const CursorNode = ({ nodeId }) => {
-    const { dispatch, state } = globalState;
-
-    const [validHoverClassLabel, setValidHoverClassLabel] = useState("");
-    const [cursor, setCursor] = useState(<CursorNode nodeId={nodeId} />);
-
-    const selectedOutputClassLabel = nodeId === state?.selectedOutput ? "selected-output" : "";
-
-    const handleMouseEnter = () => {
-        if (direction === NODE.OUTPUT) {
-            setValidHoverClassLabel(state?.selectedOutput === null ? "valid-hover" : "");
-        }
-        if (direction === NODE.INPUT) {
-            setValidHoverClassLabel(state?.selectedOutput !== null ? "valid-hover" : "")
+const InputNode = ({ nodeId, selectedOutput, dispatch, className, setInputs }) => {
+    const handleMouseUp = (e) => {
+        if (selectedOutput !== null) {
+            dispatch({
+                type: 'ADD_CONNECTION',
+                payload: { end: `${nodeId}`, setInputs },
+            });
         }
     };
 
+    return (
+        <div
+            className={className}
+            onMouseUp={handleMouseUp}
+            id={`${nodeId}`}
+        >{'>'}</div>
+    );
+};
 
-
+const OutputNode = ({ nodeId, dispatch, className }) => {
     const handleMouseDown = (e) => {
-        e.preventDefault();  // catch the mouse down here
-
+        e.preventDefault(); // catch the mouse down here
+        const CURSOR_SHIFT_HACK = 100;
         const rect = e.target.getBoundingClientRect();
         /* get the x, y offset from the center of the node */
-        const offsetX = e.clientX - rect.left - ((rect.right - rect.left) / 2);
-        const offsetY = e.clientY - rect.top - ((rect.bottom - rect.top) / 2);
-
-        if (direction === NODE.OUTPUT) {
-            dispatch({
-                type: 'START_CURSOR_NODE_DRAG',
-                payload: {
-                    start: `node-${nodeId}`,
-                    end: `cursor-node-${nodeId}`,
-                    offsetX,
-                    offsetY
-                }
-            })
-        }
+        const offsetX = (e.clientX - rect.left - (rect.right - rect.left) / 2) + CURSOR_SHIFT_HACK;
+        const offsetY = (e.clientY - rect.top - (rect.bottom - rect.top) / 2) + CURSOR_SHIFT_HACK;
+
+        dispatch({
+            type: 'START_CURSOR_NODE_DRAG',
+            payload: {
+                start: `${nodeId}`,
+                end: `cursor-${nodeId}`,
+                offsetX,
+                offsetY,
+            },
+        });
     };
 
 
-    const handleMouseUp = (e) => {
-        if ((direction === NODE.INPUT) && (state?.selectedOutput !== null)) {
-            dispatch({
-                type: 'ADD_CONNECTION',
-                payload: { end: `node-${nodeId}` }
-            })
-        }
+    return (
+        <>
+            <div className={className} id={`${nodeId}`}>{'>'}</div>
+            <CursorNode nodeId={nodeId} handleMouseDown={handleMouseDown} />
+        </>
+    );
+};
+
+const Node = ({ direction, setNodeId, setInputs }) => {

```

Comment:
Add PropTypes and clarify what these required props are doing with better variable names:

direction: a node can be an input node or an output node
setNodeId: this is a react hook created by the parent component.  It is used to establish connections between nodes.
setInputs: required if a node is an input node, otherwise not used.  This is the hook that is passed back to another component that would like to connect its output node to this input node.  This is done through the store.